src/AWS/Orleans.Streaming.SQS/Streams/SqsStreamOptions.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System.Collections.Generic;
+using Microsoft.Extensions.Options;
+
+namespace Orleans.Hosting
+{
+    public class SqsStreamOptions : PersistentStreamOptions
+    {
+        public string DeploymentId { get; set; }
+
+        public string DataConnectionString { get; set; }
+
+        public int CacheSize { get; set; } = CacheSizeDefaultValue;
+        public const int CacheSizeDefaultValue = 4096;
+
+        public int NumQueues { get; set; } = NumQueuesDefaultValue;
+        public const int NumQueuesDefaultValue = 8; // keep as power of 2.
+
+    }
+
+    public class SqsStreamOptionsFormatterResolver : IOptionFormatterResolver<SqsStreamOptions>
+    {
+        private IOptionsSnapshot<SqsStreamOptions> optionsSnapshot;
+
+        public SqsStreamOptionsFormatterResolver(IOptionsSnapshot<SqsStreamOptions> optionsSnapshot)
+        {
+            this.optionsSnapshot = optionsSnapshot;
+        }
+
+        public IOptionFormatter<SqsStreamOptions> Resolve(string name)
+        {
+            return new Formatter(name, optionsSnapshot.Get(name));
+        }
+
+        public class Formatter : IOptionFormatter<SqsStreamOptions>
+        {
+            private SqsStreamOptions options;
+            private readonly PersistentStreamOptionsFormatter parentFormatter;
+
+            public string Name { get; }
+
+            public Formatter(string name, SqsStreamOptions options)
+            {
+                this.options = options;
+                this.Name = OptionFormattingUtilities.Name<SqsStreamOptions>(name);
+                this.parentFormatter = new PersistentStreamOptionsFormatter(options);
+            }
+
+            public IEnumerable<string> Format()
+            {
+                List<string> formatted = this.parentFormatter.FormatSharedOptions();
+                formatted.AddRange(new[]
+                {
+                    OptionFormattingUtilities.Format(nameof(this.options.DeploymentId), this.options.DeploymentId),
+                    OptionFormattingUtilities.Format(nameof(this.options.DeploymentId), "<-Redacted->"),
```

Comment:
Error here, duplicated line. I think the latter is supposed to be `DataConnectionString`. We can use the existing `ConfigUtils.RedactConnectionStringInfo` method as in other providers - unless you feel it isn't adequate?

Reply:
Good catch on the dupe, tnx.
The redact utility, by my understanding, expects an azure connection string, and knows how to pull the secrets out of that.  I don't know if we can trust it to redact secrets from other service connection strings.  It would be valuable to be able to strip out secrets for all of these, but I don't have the time to investigate each format, so I've been just placing the generic <-redact-> line in there.

Reply:
That's quite fair - I simply don't know how to ensure that we can redact SQS keys correctly, or if we even should. Redacting the entire thing is safer

Reply:
Looking at the RedactConnectionStringInfo logic, I'm not confident it's safe.  If a connection string has one secret in the list, and one not in the list, it's secrets will be logged.  I'm going to stick with the <--SNIP--> for now.  Developers/maintainers of a component can develop component specific redactors if it's important.  IMO it's better to log less info than risk logging users secrets.